src/components/SignIn/SignIn.tsx

```diff
@@ -17,33 +17,55 @@ import styles from './SignIn.module.scss';
 
 export type SignInProps = Omit<ModalProps, 'overlay' | 'zIndex'>;
 
-let emailValue: string = '';
+export function sessionReducer(state: any, action: any): any {
+  switch (action.type) {
+    case "SIGNIN":
+      return {
+         ...state,
+	sessionKey: action.sessionKey,
+      };
+    case "LOGOUT":
+      return {
+        ...state,
+	sessionKey: null,
+      };
+    default:
+      return state;
+  }
+};
+export let sessionKey: string;// TODO store the sessionKey in a nicer way
 
-function emailSignIn(event: React.FormEvent) {
-  const requestOptions = {
-        method: 'POST',
-        headers: { 'Content-Type': 'application/json' },
-        body: JSON.stringify({ email: emailValue })
-    };
+export function SignIn(props: SignInProps): React.ReactElement {
+  let emailValue: string = '';
+
+  const [sessionState, sessionDispatch] = React.useReducer(sessionReducer, null);// null for initial state
```

Comment:
The comment is redundant. But even then, you should pass an empty object instead of null.